src/core/tests/frontend/paddle/test_models/gen_scripts/generate_gather.py

```diff
@@ -0,0 +1,108 @@
+#
+# gather paddle model generator
+#
+import numpy as np
+from save_model import saveModel
+import paddle
+import sys
+
+
+def gather1(name: str, x, y):
+    paddle.enable_static()
+
+    with paddle.static.program_guard(paddle.static.Program(), paddle.static.Program()):
+        data = paddle.static.data(name='x', shape=x.shape, dtype=x.dtype)
+        index = paddle.static.data(name='index', shape=y.shape, dtype=y.dtype)
+        out = paddle.gather(data, index)
+        if x.dtype == "int64":
+            out = paddle.cast(out, "float32")
+        out = paddle.cast(out, "float32")
+
+        cpu = paddle.static.cpu_places(1)
+        exe = paddle.static.Executor(cpu[0])
+        # startup program will call initializer to initialize the parameters.
+        exe.run(paddle.static.default_startup_program())
+
+        outs = exe.run(
+            feed={'x': x, 'index': y},
+            fetch_list=[out])
+
+        saveModel(name, exe, feedkeys=['x', 'index'], fetchlist=[out], inputs=[
+                  x, y], outputs=[outs[0]], target_dir=sys.argv[1])
+
+    return outs[0]
+
+# def gather2(name:str, x, y, z):
```

Comment:
Why did you add new code with some commented parts?

Reply:
this commented test case is for Paddle 'gather' op with 'axis' as common input case, while currently OpenVINO only support Constant operation on 'axis' input(https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/mkldnn_gather_node.cpp#L31) . so I think this commented test case together with commented testcase name in 'fuzzy test' list(https://github.com/openvinotoolkit/openvino/blob/347a3bd8a227a88e70487104e1ea3739f5597482/src/core/tests/frontend/paddle/op_fuzzy.cpp#L140) may help us notice this op limitation and will easily remove this limitation when our OpenVINO support this feature.

Reply:
> @liubo-intel Can we merge the 2 functions to remove the code redundant?
> 
> The spec has no such limitation. Checked mkldnn plugin, looks the limitation could be removed. Could you please have a try?
> 
> @nshchego Could you please have a look?https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/mkldnn_gather_node.cpp#L31

hi, @ceciliapeng2011 : have merge the 2 functions. while it looks like the mkldnn plugin limitation is still exist

Reply:
@liubo-intel Have you tried to remove https://github.com/openvinotoolkit/openvino/blob/714d7a79c73da1f6932f68fdf736c8acd865f5ca/src/plugins/intel_cpu/src/nodes/gather.cpp#L33 and tested it works with non-const 'axis' input? If it works, could you include it in PR and ask cpu plugin owners review? 
@nshchego There are cases that 'axis' input for static node. I checked the algo of gather node here, looks we can remove this restriction.

Reply:
Axis value affects output shape, thus it must be a constant for node with static shapes, otherwise such shape could not be calculated properly before inference.
If Axes input is not a constant, node output must have dynamic shape. In this case shape will be calculated on each inference with actual axes value.